test/pstl_offload/memory/usm_memory_replacement_native_alloc.cpp

```diff
@@ -0,0 +1,23 @@
+// -*- C++ -*-
+//===----------------------------------------------------------------------===//
+//
+// Copyright (C) 2023 Intel Corporation
+//
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+//===----------------------------------------------------------------------===//
+
+#if !__SYCL_PSTL_OFFLOAD__
+#error "PSTL offload compiler mode should be enabled to run this test"
+#endif
+
+#define DO_STD_INJECTION 1
+#include "system_allocations.h"
+
+void perform_system_allocations(allocs& na) {
+    perform_allocations_impl(na);
```

Comment:
I would also add a comment that this cpp file emulates system allocations while compiling with pstl offload option

Reply:
Done.